### DISCUSSION TITLE
How to delete the vectorstore?

### DISCUSSION THREAD

**Discussion by madeganesh22** [imartinez/privateGPT#779] [Q&A] [answered]
I ran ingest.py and deleted a file from the source_documents folder. Then, I ran ingest.py again. However, when I ran privateGPT.py and entered a query, it gave me an answer from the deleted file. How can I delete it permanently?

**Comment by andyjain** [comment, marked as answer]
The way I know is to delete the db folder and then ingest again.

**Reply by madeganesh22** [reply]
It works, thanks!

**Reply by mylee04** [reply]
But then, every time I add or delete some documents, do I need to run ingest.py again? Isn't that inefficient? If I have 100 documents and delete a few of them, I would have to run the script again. It would take a lot of time every time if I need to run ingest.py. Do you know other ways or idea to make ingest only changes like new files or deleted documents lists?

**Reply by Mrflooglebinder** [reply]
aggree with Kevin on this, it would be ideal if it could recognise file changes and only re-ingest new and remove redundant data. Got the answer I need for now though thanks, Awesome project BTW. Big thumbs up!
